surveyMock.go

```diff
@@ -0,0 +1,84 @@
+package survey
+
+import (
+	"io"
+
+	"github.com/AlecAivazis/survey/v2/core"
+	"github.com/AlecAivazis/survey/v2/terminal"
+)
+
+type SurveyMock struct {
+	singleResponse    interface{}
+	multipleResponses map[string]interface{}
+}
+
+// Not implemented, because it does not affect the mock
+func (mock *SurveyMock) WithStdio(in terminal.FileReader, out terminal.FileWriter, err io.Writer) AskOpt {
```

Comment:
Is there any reason to keep these options funcs at the Survey interface level? Calling them as part of the interface has no utility: they can't be mocked, they can't be verified, and they only forward the call to the funcs on the package level. Could we just keep them on the package level instead?

Reply:
There are pros and cons to keeping all the functions in the interface:

Pro: 
- Less confusing to have to call some functions on the struct and some on the package#
- Allows for stateful functions in the future

Con:
- It convolutes the library code and forces us to maintain a lot of function declarations in three places instead of two

I'm not sure which aspect is more important, but I would gravitate to keeping them in the interface. 
However, you have the final word 😄

Reply:
I thought about it some more. 

I think the two pros I listed are either weak arguments (number 1) or can still be implemented later, even if we don't add them now (number two). 

I'll remove all the functions except Ask/AskOne from the interface.